File: lanzou.py
#!/usr/bin/env python

import requests
import json

UA="Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4537.0 Safari/537.36 Edg/93.0.926.1"
header={"user-agent":UA}

def swn(str:str,start:str,end:str)->str:
    r=str.find(start)
    re=str.find(end,r+len(start))
    return str[r+len(start):re-len(end)+1]

def file(file,key=""):
    html=requests.get(file,headers=header)
    if html.text.find("输入密码")!=-1 and key=="":
        return "请输入密码!"
    if key=="":
        start=html.text.find("src=\"")
        start=html.text.find("src=\"",start+6) #找到目标
        end=html.text.find("\"",start+6)
        html=requests.get(url="https://xihale.lanzoui.com/"+html.text[start+6:end])
        data=html.text
        start=data.find("var ajaxdata = '")
        end=data.find("'",start+16)
        ajaxdata=data[start+16:end]
        start=data.find("'",end+3)
        end=data.find("'",start+1)
        postdown=data[start+1:end]
    else:
        start=html.text.find("&sign=")
        postdown=html.text[start+6:html.text.find("&p=",start+6)]

    html=requests.post(url="https://lanzoux.com/ajaxm.php",headers={"referer":html.url,"user-agent":UA},data={"action":"downprocess","sign":postdown,"p":key})
    data=json.loads(html.text)
    if data["url"]==0:
        return data["inf"]
    html=requests.get(url=data["dom"]+"/file/"+data["url"],headers={
        'Accept':'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8',
        'Accept-Language': 'zh-CN,zh;q=0.8',
        'Connection': 'keep-alive',},allow_redirects=False)
    return html.headers['Location'] #获取重定向网址

def filer(filer,key=""):
    html=requests.get(filer,headers=header)
    r=html.text.find("data : {")+7
    re=html.text.find("}",r)
    text=html.text[r:re]+'}'
    text=text.strip()
    ib=swn(html.text,"'t':",",")
    ibs=swn(html.text,ib+" = '","'")
    ih=swn(html.text,"'k':",',')
    ihs=swn(html.text,ih+" = '","'")
    text=text.replace(ib,"'"+ibs+"'")
    text=text.replace(ih,"'"+ihs+"'")
    text=text.replace("pgs","'1'")
    text=text.replace("'",'"')
    if key!="":
        text=text.replace(":pwd",':"'+key+'"')
    else:
        l=list(text)
        l.pop(text.rfind(","))
        text=''.join(l)
    # print(text)
    text=json.loads(text)
    # print(text)
    html=requests.post("https://lanzoux.com/filemoreajax.php",headers={"referer":html.url,"user-agent":UA},data=text)
    
    data=json.loads(html.text)
    if data["info"]!="sucess":
        return data["info"]
    return data

# print(filer("https://xihale.lanzoui.com/b015tdsbi"))

# print(filer("https://xihale.lanzoui.com/b016811pi","xihale"))

# print(json.dumps("{a:1,b:2}"))

File: lanzou.js
const lanzou='https://lanzoux.com';
function ajax(url,data=null,headers={},type='POST'){
    var r;
    $.ajax({url:url,data:data,type:type,headers:headers,async:false,success:(result)=>{
        r=result;
    }});
    return r;
}
function lanzouGet(url,pwd=''){
    if(url.match(/^http/)===null)url=`${lanzou}/${url}`;
    const maker=function(a,b){return {code:a,info:b};},swn=function(str,start,end){s=str.indexOf(start);return str.substring(s+start.length,str.indexOf(end,s+1));};
    let data,file=ajax(url,null,{},"GET");
    if(file.indexOf("文件不存在，或已删除")!=-1)return maker(0,"文件不存在，或已删除");
    if(file.indexOf("文件受密码保护，请输入密码继续下载")!=-1&&pwd=='')
        while(pwd==null||pwd=="")
            pwd=prompt("需要密码","");
    if(file.indexOf("filemoreajax.php")!=-1){ //filer
        pgs=1;
        eval(swn(file,"var pgs;","pgs =1;"));
        eval('data={'+swn(file,"data : {","}")+'}');
        return maker(2,ajax(`${lanzou}/filemoreajax.php`,data));
    }else{ // file
        let start=file.indexOf('/',file.lastIndexOf('e>-',file.length-1062));
        file=ajax(lanzou+file.substring(start,file.indexOf('"',start+1)),null,{},"GET");
        eval("var"+swn(file,'var','$'));
        eval(`data=${swn(file,'data : ','}')}}`);
        data=ajax(`${lanzou}/ajaxm.php`,data);
        return maker(1,`${data.dom}/file/${data.url}`);
    }
}
// file('https://xihale.lanzoui.com/iAVdonx0egb');
// console.log(lanzouGet('https://xihale.lanzoui.com/iAVdonx0egb'));